modules/47_regipol/regiCarbonPrice/bounds.gms

```diff
@@ -18,36 +18,116 @@
 loop(regi$(sameAs(regi,"DEU")),
   vm_deltaCap.up("2025",regi,"wind","1")=2*smax(tall$(tall.val ge 2011 and tall.val le 2020), pm_delta_histCap(tall,regi,"wind"));
   vm_deltaCap.up("2025",regi,"spv","1")=2*smax(tall$(tall.val ge 2011 and tall.val le 2020), pm_delta_histCap(tall,regi,"spv"));
+  vm_cap.up("2025",regi,"spv","1")=0.12;
 );
+$endIf.tech_bounds_2025
+
 
 *' These bounds account for historic gas power development.
-vm_capTotal.up("2015",regi,"pegas","seel")$(sameas(regi,"DEU"))=30/1000;
-vm_capTotal.up("2020",regi,"pegas","seel")$(sameas(regi,"DEU"))=34/1000;
+v47_prodSEtotal.up("2020",regi,"pegas","seel")$(sameAs(regi,"DEU"))= 0.36*sm_EJ_2_TWa;
+$ifThen.tech_bounds_2025 "%cm_tech_bounds_2025%" == "on"
+v47_prodSEtotal.up("2025",regi,"pegas","seel")$(sameAs(regi,"DEU"))= 0.4*sm_EJ_2_TWa;
+$endIf.tech_bounds_2025
 
 *' These bounds account for historic coal power development.
 vm_cap.up("2020",regi,"pc","1")$((cm_startyear le 2020) and (sameas(regi,"DEU"))) = 38.028/1000;
+*' This limits early retirement of coal power in Germany in 2020s to avoid extremly fast phase-out.
+vm_capEarlyReti.up('2025',regi,'pc')$(sameAs(regi,"DEU")) = 0.65;
 
-*' These bounds account for historic and near-term gas power development and prevent too sudden coal to gas switch in Germany by 2025.
+
+*' This aligns 2020 chp capcities for Germany with historic data (AGEB)
+*' most of district heating is provided by CHP plants.
+*' coal share of chp heat output to be between 20-25% of total district heating demand
+*' gas share of chp heat output to be between 50-55% of total district heating demand
+*' bio share of chp heat output to be between 15-25% of total district heating demand
 loop(regi$(sameAs(regi,"DEU")),
-vm_deltaCap.up("2015",regi,"ngcc","1") = 0.002;
-vm_deltaCap.up("2020",regi,"ngcc","1") = 0.0015;
-vm_deltaCap.up("2025",regi,"ngcc","1") = 0.0015;
-*' Along the same lines, this limits early retirement of coal power in Germany in 2020s to avoid extremly fast phase-out.
-vm_capEarlyReti.up('2025',regi,'pc') = 0.65; 
+    loop(t$(t.val eq 2020),
+        vm_cap.lo(t,regi,"coalchp","1")= 0.2
+*** total FE heat demand from industry and buildings calibration trajectories
+                                            * (pm_cesdata(t,regi,"feheb","quantity")
+                                                + pm_cesdata(t,regi,"feheb","quantity"))
+*** divide by tdhes efficiency get from FE heat to SE heat, i.e. take into account heat transmission losses
+                                            / pm_eta_conv(t,regi,"tdhes")
+*** divide by pm_prodCouple to get from heat production to production of seel
+                                            /  pm_prodCouple(regi,"pecoal","seel","coalchp","sehe")
+*** divide by capacity factor to get from seel production in TWa to capacity in TW
+                                            / pm_cf(t,regi,"coalchp");
+
```

Comment:
Couldn't you create a more generic parameter to avoid the code repetition in these lines.
Something like `pm_cesCap(t,regi,te,rlf)` or something similar, that is calculated in the core and just used here to clean a bit this code?

Reply:
Added a note. Should be done via mrremind more cleanly.